lib/cloudregister/registerutils.py

```diff
@@ -109,19 +109,23 @@ def check_registration(smt_server_name):
 # ----------------------------------------------------------------------------
 def clean_hosts_file(domain_name):
     """Remove the smt server entry from the /etc/hosts file"""
+    if isinstance(domain_name, str):
+        domain_name = domain_name.encode()
     new_hosts_content = []
-    content = open(HOSTSFILE_PATH, 'r').readlines()
+    # Handle entries as bytes,
+    # Yes, users put non ascii characters into /etc/hosts
+    content = open(HOSTSFILE_PATH, 'rb').readlines()
     smt_announce_found = None
     for entry in content:
-        if '# Added by SMT' in entry:
+        if b'# Added by SMT' in entry:
             smt_announce_found = True
             continue
         if smt_announce_found and domain_name in entry:
             smt_announce_found = False
             continue
         new_hosts_content.append(entry)
 
-    with open(HOSTSFILE_PATH, 'w') as hosts_file:
+    with open(HOSTSFILE_PATH, 'wb') as hosts_file:
```

Comment:
Using a context manager here but the previous usage is not (line: 117). Therefore the file could still be open for reading and now be open a second time for writing. Suggest using a context manager every time the file is open to ensure it's closed when processing is finished.